paddle/fluid/operators/conv_mkldnn_op.cc

```diff
@@ -386,14 +461,24 @@ class ConvMKLDNNGradOpKernel : public paddle::framework::OpKernel<T> {
     auto chosen_memory_format =
         platform::data_format_to_memory_format(data_format);
 
+    weights_format =
+        (g == 1) ? chosen_memory_format : mkldnn::memory::format::goihw;
+
+    if (is_conv3d) {
+      chosen_memory_format =
+          platform::MKLDNNFormatForSize(src_tz.size(), chosen_memory_format);
+      weights_format =
+          (g == 1) ? chosen_memory_format : mkldnn::memory::format::goidhw;
+    }
```

Comment:
line 464-472 are same with line 149-157. please make a function and reuse it.

Reply:
Done